tutorials/01-gaussian-mixture-model/01_gaussian-mixture-model.jmd

```diff
@@ -135,30 +136,85 @@ After sampling we can visualize the trace and density of the parameters of inter
 We consider the samples of the location parameters $\mu_1$ and $\mu_2$ for the two clusters.
 
 ```julia
-plot(chains[["μ[1]", "μ[2]"]]; colordim=:parameter, legend=true)
+plot(chains[["μ[1]", "μ[2]"]], legend=true)
 ```
 
 It can happen that the modes of $\mu_1$ and $\mu_2$ switch between chains.
-For more information see the [Stan documentation](https://mc-stan.org/users/documentation/case-studies/identifying_mixture_models.html) for potential solutions.
+For more information see the [Stan documentation](https://mc-stan.org/users/documentation/case-studies/identifying_mixture_models.html). This is because it's possible for either model parameter $\mu_k$ to be assigned to either of the corresponding true means, and this assignment need not be consistent between chains. 
 
-We also inspect the samples of the mixture weights $w$.
+That is, the posterior is fundamentally multimodal, and different chains can end up in different modes, complicating inference.
+
+One solution here is to enforce an ordering on our $\mu$ vector, requiring $\mu_k > \mu_{k-1}$ for all $k$. 
+
+`Bijectors.jl` [provides](https://turinglang.org/Bijectors.jl/dev/transforms/#Bijectors.OrderedBijector) an easy transformation (`ordered()`) for this purpose:
+
+```julia
+using Bijectors: ordered
+
+@model function gaussian_mixture_model_ordered(x)
+    # Draw the parameters for each of the K=2 clusters from a standard normal distribution.
+    K = 2
+    μ ~ ordered(MvNormal(Zeros(K), I))
+
+    # Draw the weights for the K clusters from a Dirichlet distribution with parameters αₖ = 1.
+    w ~ Dirichlet(K, 1.0)
+    # Alternatively, one could use a fixed set of weights.
+    # w = fill(1/K, K)
+
+    # Construct categorical distribution of assignments.
+    distribution_assignments = Categorical(w)
+
+    # Construct multivariate normal distributions of each cluster.
+    D, N = size(x)
+    distribution_clusters = [MvNormal(Fill(μₖ, D), I) for μₖ in μ]
+
+    # Draw assignments for each datum and generate it from the multivariate normal distribution.
+    k = Vector{Int}(undef, N)
+    for i in 1:N
+        k[i] ~ distribution_assignments
+        x[:, i] ~ distribution_clusters[k[i]]
+    end
+
+    return k
+end
+
+model = gaussian_mixture_model_ordered(x);
+```
+
+Now, re-running our model, we can see that the assigned means are consistent across chains:
+
+```julia
+chains = sample(model, sampler, MCMCThreads(), nsamples, nchains; discard_initial = burn);
```

Comment:
Maybe let's keep the tutorial simple and avoid surprising warnings in singlethreaded environments:
```suggestion
chains = sample(model, sampler, nsamples, nchains; discard_initial = burn);
```

Reply:
Actually un-resolving this because I don't think it works? As it is right now, I'm not sure if Turing allows multiple chains without specifying a type of parallelism. 

[The documentation](https://turinglang.org/v0.30/docs/using-turing/guide/#sampling-multiple-chains), if it's current, seems to suggest I should do something like:

```julia
chains = mapreduce(c -> sample(model_fun, sampler, 1000), chainscat, 1:num_chains)
```

I'm not sure if that's worth it just to get rid of the warning  — let me know what you think though.